Prepare for 6.6.0 release (#652)
gazebosim/gz-physics: Changelog.md

## Gazebo Physics 6.x

### Gazebo Physics 6.6.0 (2024-06-11)

> 1. dartsim: optimize picking contact points with ODE collision detectorq

Reviewer: ```suggestion
1. dartsim: optimize picking contact points with ODE collision detector
```

    * [Pull request #584](https://github.com/gazebosim/gz-physics/pull/584)

1. Fix windows compiler warning
    * [Pull request #629](https://github.com/gazebosim/gz-physics/pull/629)

1. Disable test failing due to ODE/libccd
    * [Pull request #621](https://github.com/gazebosim/gz-physics/pull/621)

1. bullet-featherstone: Improve mesh collision stability
    * [Pull request #600](https://github.com/gazebosim/gz-physics/pull/600)

1. bullet-featherstone: Support nested models
    * [Pull request #574](https://github.com/gazebosim/gz-physics/pull/574)

1. Revert "bazel: updates for garden (#513)"

>     * [Pull request #513)"](https://github.com/gazebosim/gz-physics/pull/513)")

Reviewer: ```suggestion
    * [Pull request #513](https://github.com/gazebosim/gz-physics/pull/513)
```

1. Garden test cleanup
    * [Pull request #587](https://github.com/gazebosim/gz-physics/pull/587)

1. Support setting max contacts in dartsim's ODE collision detector
    * [Pull request #582](https://github.com/gazebosim/gz-physics/pull/582)

1. Get bullet version from cmake instead of API
    * [Pull request #591](https://github.com/gazebosim/gz-physics/pull/591)

1. Reduce error to debug messsage for mesh construction (#581)

>     * [Pull request #581)(https://github.com/gazebosim/gz-physics/pull/581)

Reviewer: ```suggestion
    * [Pull request #581](https://github.com/gazebosim/gz-physics/pull/581)
```

1. bullet-featherstone: Set collision spinning friction
    * [Pull request #579](https://github.com/gazebosim/gz-physics/pull/579)

1. Infrastructure
    * [Pull request #578](https://github.com/gazebosim/gz-physics/pull/578)
    * [Pull request #572](https://github.com/gazebosim/gz-physics/pull/572)

1. dartsim: fix handling inertia matrix pose rotation
    * [Pull request #351](https://github.com/gazebosim/gz-physics/pull/351)

1. bullet-featherstone: fix setting angular velocity
    * [Pull request #567](https://github.com/gazebosim/gz-physics/pull/567)

1. bullet-featherstone: support off-diagonal inertia
    * [Pull request #544](https://github.com/gazebosim/gz-physics/pull/544)

> 1. bullet-featherstoneFix how links are flattened in  ConstructSdfModel

Reviewer: ```suggestion
1. bullet-featherstone: Fix how links are flattened in  ConstructSdfModel
```

## Gazebo Physics 6.x

### Gazebo Physics 6.6.0 (2024-06-11)

1. dartsim: optimize picking contact points with ODE collision detectorq
    * [Pull request #584](https://github.com/gazebosim/gz-physics/pull/584)

1. Fix windows compiler warning
    * [Pull request #629](https://github.com/gazebosim/gz-physics/pull/629)

1. Disable test failing due to ODE/libccd
    * [Pull request #621](https://github.com/gazebosim/gz-physics/pull/621)

1. bullet-featherstone: Improve mesh collision stability
    * [Pull request #600](https://github.com/gazebosim/gz-physics/pull/600)

1. bullet-featherstone: Support nested models
    * [Pull request #574](https://github.com/gazebosim/gz-physics/pull/574)

1. Revert "bazel: updates for garden (#513)"
    * [Pull request #513)"](https://github.com/gazebosim/gz-physics/pull/513)")

1. bazel: updates for garden

>     * [Pull request #513](https://github.com/gazebosim/gz-physics/pull/513)

Reviewer 2: This can be removed

Author: Removed in [804a3ef](https://github.com/gazebosim/gz-physics/pull/652/commits/804a3ef4532f1b0000686ea8f0aca56b3f9780de)

1. Garden test cleanup
    * [Pull request #587](https://github.com/gazebosim/gz-physics/pull/587)

1. Support setting max contacts in dartsim's ODE collision detector
    * [Pull request #582](https://github.com/gazebosim/gz-physics/pull/582)

1. Get bullet version from cmake instead of API
    * [Pull request #591](https://github.com/gazebosim/gz-physics/pull/591)

1. Prepare for 2.6.2 release

>     * [Pull request #585](https://github.com/gazebosim/gz-physics/pull/585)

Reviewer 2: Remove

Author: Removed in [804a3ef](https://github.com/gazebosim/gz-physics/pull/652/commits/804a3ef4532f1b0000686ea8f0aca56b3f9780de)

1. Reduce error to debug messsage for mesh construction (#531)

>     * [Pull request #531) (#581](https://github.com/gazebosim/gz-physics/pull/531) (#581)

Reviewer 2: I think this was a backport. We can keep the original PR number

Author: Replaced with #581 instead [804a3ef](https://github.com/gazebosim/gz-physics/pull/652/commits/804a3ef4532f1b0000686ea8f0aca56b3f9780de)

1. bullet-featherstone: Set collision spinning friction
    * [Pull request #579](https://github.com/gazebosim/gz-physics/pull/579)

1. CI workflow: use 22.04 for codecheck, coverage
    * [Pull request #578](https://github.com/gazebosim/gz-physics/pull/578)

1. Update github action workflows

>     * [Pull request #572](https://github.com/gazebosim/gz-physics/pull/572)

Reviewer 2: This can be consolidated into "Infrastructure"

Author: Consolidated in [804a3ef](https://github.com/gazebosim/gz-physics/pull/652/commits/804a3ef4532f1b0000686ea8f0aca56b3f9780de)
